### ISSUE TITLE
Build number is generating wrong.

### ISSUE THREAD

**Issue by oursridhar** [GitTools/GitVersion#1615]
Here with i have attached logs ( version.log & yaml file generated by Gitversion 4.0.0"

Build number suppose to be 2.16.0.*********, However it is giving build number as 6.5.0.************

could some one help me on this. 

2019-03-01 08:57:41		INFO [03/01/19 8:57:41:16] Working directory: .
2019-03-01 08:57:41		INFO [03/01/19 8:57:41:47] Begin: Calculating base versions
2019-03-01 08:57:41		  INFO [03/01/19 8:57:41:91] Fallback base version: 0.1.0 with commit count source 560f003c8bc72132a496ea12b6196e96740ec6cb (Incremented: None)
2019-03-01 08:57:42		  INFO [03/01/19 8:57:42:87] Git tag '2.15.0-rc': 2.15.0-rc with commit count source 92ceb82c527ae7f44ae0dd1cf13d9a60cd0d7557 (Incremented: 2.15.0-rc)
2019-03-01 08:57:42		  INFO [03/01/19 8:57:42:89] Git tag '2.15.0': 2.15.0 with commit count source a3b36f14fb7348ae2d06b9c234a7dc46be76c5fe (Incremented: 2.16.0)
2019-03-01 08:57:42		  INFO [03/01/19 8:57:42:92] Git tag '2.10.0': 2.10.0 with commit count source dbd832560f61a6730561c24b3e61a0dc757a19a8 (Incremented: 2.11.0)
2019-03-01 08:57:42		  INFO [03/01/19 8:57:42:94] Git tag '2.10.0-rc': 2.10.0-rc with commit count source dbd832560f61a6730561c24b3e61a0dc757a19a8 (Incremented: 2.10.0-rc)
2019-03-01 08:57:43		  INFO [03/01/19 8:57:43:47] Merge message 'Merge pull request #1655 in from release/2.15.0 to master

* commit '92ceb82c527ae7f44ae0dd1cf13d9a60cd0d7557': (871 commits)
  DE17276-Updated the "lastUpdatedBy" variable and added command to update the added data.
  ...': 2.15.0 with commit count source b16ea45f6a19ccaa553b34ae5214cd222c48d070 (Incremented: 2.16.0)
2019-03-01 08:57:43		  INFO [03/01/19 8:57:43:51] Merge message 'Merge pull request #1544 in  from release/2.10.0 to master

Merging Release 2.10.0 into Master

* commit 'dbd832560f61a6730561c24b3e61a0dc757a19a8': (291 commits)
  DE15231: changed return type for unit tests
  ...': 2.10.0 with commit count source a3b36f14fb7348ae2d06b9c234a7dc46be76c5fe (Incremented: 2.11.0)
2019-03-01 08:57:43		  INFO [03/01/19 8:57:43:54] Merge message 'Merge pull request #1282 in  from DE15280-Fix-flaky-tests-6.4.18.10-11-12 to develop

* commit 'ee187686eb26be8ab9e458c35520b3281414362b':
  DE15280: Revert routing change and changed components to use a subscribe for getting permissions
  DE15280: Readded outer resolver but changed to the sync version so it can't flush out the results of the resolver from the inner. Both will fire on an inner route.
  DE15280: Repair timing issue with tasks by using the sync resolver rather than the async one': 6.4.18+10 with commit count source ac735e3da73a22a66fb00ec000a83c53444ea1e3 (Incremented: 6.5.0)
2019-03-01 08:57:43		  INFO [03/01/19 8:57:43:60] Merge message 'Merge pull request #747 in  from DE12244-6.2_Setup_Feature_Group_tags_update to develop

* commit '585017f770c603616c2efc1d81d06ee5240bd8aa':
  DE12244: Updated FeatureGroup tags and changed names for UX feature files 6.2.31, 6.2.32, 6.2.33. Updated FeatureGroup tag for service feature file 6.2.3': 6.2.0 with commit count source 45308fddd9b7b632cb86be0f1bda77fceeab1d81 (Incremented: 6.3.0)
2019-03-01 08:57:43		  INFO [03/01/19 8:57:43:68] Merge message 'Merge pull request #748 in  from DE12263-6.4_Operations_Feature_Group_tags_update to develop

* commit '59cd50acbe0c867bb3d65eb40c0954d65351ad4d':
  DE12263: Updated FeatureGroup tag for the service feature file 6.4.3': 6.4.0 with commit count source 732b89617370c58cddb73b1bc6a9c9a723f4468c (Incremented: 6.5.0)
2019-03-01 08:57:43		  INFO [03/01/19 8:57:43:73] Found multiple base versions which will produce the same SemVer (6.5.0), taking oldest source for commit counting (Merge message 'Merge pull request #748 in  from DE12263-6.4_Operations_Feature_Group_tags_update to develop

* commit '59cd50acbe0c867bb3d65eb40c0954d65351ad4d':
  DE12263: Updated FeatureGroup tag for the service feature file 6.4.3')
2019-03-01 08:57:43		  INFO [03/01/19 8:57:43:77] Base version used: Merge message 'Merge pull request #748 in from DE12263-6.4_Operations_Feature_Group_tags_update to develop

* commit '59cd50acbe0c867bb3d65eb40c0954d65351ad4d':
  DE12263: Updated FeatureGroup tag for the service feature file 6.4.3': 6.4.0 with commit count source 732b89617370c58cddb73b1bc6a9c9a723f4468c (Incremented: 6.5.0)
2019-03-01 08:57:43		INFO [03/01/19 8:57:43:77] End: Calculating base versions (Took: 2,304.35ms)
2019-03-01 08:57:43		INFO [03/01/19 8:57:43:97] 11252 commits found between 732b89617370c58cddb73b1bc6a9c9a723f4468c and 340e3df41e6bfb5b28c4c0dac2c23eb661574100
2019-03-01 08:57:43		INFO [03/01/19 8:57:43:98] Begin: Getting version tags from branch 'refs/heads/develop'.
2019-03-01 08:57:44		INFO [03/01/19 8:57:44:27] End: Getting version tags from branch 'refs/heads/develop'. (Took: 283.21ms)
2019-03-01 08:57:44		INFO [03/01/19 8:57:44:31] Begin: Creating dictionary
2019-03-01 08:57:44		INFO [03/01/19 8:57:44:32] End: Creating dictionary (Took: 4.06ms)
2019-03-01 08:57:44		INFO [03/01/19 8:57:44:32] Begin: Storing version variables to cache file D:\BuildAgent1\work\69e046729f437848\.git\gitversion_cache\ACDE664D99692F7EBE8F6D85C84333E8578D87DA.yml
2019-03-01 08:57:44		INFO [03/01/19 8:57:44:37] End: Storing version variables to cache file D:\BuildAgent1\work\69e046729f437848\.git\gitversion_cache\ACDE664D99692F7EBE8F6D85C84333E8578D87DA.yml (Took: 46.25ms)
2019-03-01 08:57:44		INFO [03/01/19 8:57:44:37] Applicable build agent found: 'TeamCity'.

And yml file
Major: 6
Minor: 5
Patch: 0
PreReleaseTag: alpha.11252
PreReleaseTagWithDash: -alpha.11252
PreReleaseLabel: alpha
PreReleaseNumber: 11252
BuildMetaData: ''
BuildMetaDataPadded: ''
FullBuildMetaData: Branch.develop.Sha.340e3df41e6bfb5b28c4c0dac2c23eb661574100
MajorMinorPatch: 6.5.0
SemVer: 6.5.0-alpha.11252
LegacySemVer: 6.5.0-alpha11252
LegacySemVerPadded: 6.5.0-alpha11252
AssemblySemVer: 6.5.0.0
AssemblySemFileVer: 6.5.0.0
FullSemVer: 6.5.0-alpha.11252
InformationalVersion: 6.5.0-alpha.11252+Branch.develop.Sha.340e3df41e6bfb5b28c4c0dac2c23eb661574100
BranchName: develop
Sha: 340e3df41e6bfb5b28c4c0dac2c23eb661574100
ShortSha: 340e3df
NuGetVersionV2: 6.5.0-alpha11252
NuGetVersion: 6.5.0-alpha11252
NuGetPreReleaseTagV2: alpha11252
NuGetPreReleaseTag: alpha11252
CommitsSinceVersionSource: 11252
CommitsSinceVersionSourcePadded: 11252
CommitDate: 2019-03-01

**Comment by marcwittke** [comment]
```
2019-03-01 08:57:43 INFO [03/01/19 8:57:43:54] Merge message 'Merge pull request #1282 in from DE15280-Fix-flaky-tests-6.4.18.10-11-12 to develop
```

is your problem. `6.4.18` is detected as version to bump. This behavior appeared in GitVersion 4.0.0 and there are several issues (#1468, #1481) and already some PRs to correct it (#1495,  #1507).

Try [Version 3.6.5](https://github.com/GitTools/GitVersion/releases/tag/v3.6.5) as temporary workaround.

**Comment by marcwittke** [comment]
in 5.0.0 it was fixed anyway
